var/spack/repos/builtin/packages/py-poetry/package.py

```diff
@@ -12,12 +12,17 @@ class PyPoetry(PythonPackage):
     homepage = "https://python-poetry.org/"
     pypi = "poetry/poetry-1.1.12.tar.gz"
 
+    version("1.2.1", sha256="2750bb2b636ef435d8beac51dde0b13d06199017a1d9b96cba899863d1e81024")
+    version("1.2.0", sha256="17c527d5d5505a5a7c5c14348d87f077d643cf1f186321530cde68e530bba59f")
     version("1.1.13", sha256="b905ed610085f568aa61574e0e09260c02bff9eae12ff672af39e9f399357ac4")
     version("1.1.12", sha256="5c66e2357fe37b552462a88b7d31bfa2ed8e84172208becd666933c776252567")
 
     depends_on("python@2.7,3.5:3", type=("build", "run"))
-    depends_on("py-poetry-core@1.0.7:1.0", type=("build", "run"))
-    depends_on("py-cleo@0.8.1:0.8", type=("build", "run"))
+    depends_on("py-poetry-core@1.0.7:1.0", type=("build", "run"), when="@:1.1")
+    depends_on("py-poetry-core@1.1.0", type=("build", "run"), when="@1.2.0:")
+    depends_on("py-poetry-plugin-export@1.0.6:1", type=("build", "run"), when="@1.2.0:")
+    depends_on("py-cleo@0.8.1:0.8", type=("build", "run"), when="@:1.1")
+    depends_on("py-cleo@1.0.0a5:", type=("build", "run"), when="@1.2.0:")
```

Comment:
```suggestion
    depends_on("py-cleo@1:", type=("build", "run"), when="@1.2.0:")
```
Spack's version comparison is broken, and 1.0.0 is considered to be < 1.0.0a5. So if you keep the logic as is, it won't apply once cleo 1.0.0 is released